### PR TITLE
Simplify MID pre-cluster

### PR DESCRIPTION
**Pull request by dstocco**
MID pre-cluster stricture is simplified so that they can be used in DPL without a (time-consuming) serialisation.
The aim is to use as much as possible trivially_copiable objects.

The MID Cluster2D object was used in a test the boost serialization.
Since we now remove the boost serialization methods from it, the test was modified.
This is done in the first of the two commits. Please do not squash the commits when merging.

### CODE REVIEW

File: Detectors/MUON/MID/Clustering/include/MIDClustering/Clusterizer.h [hunk 1]
```diff
@@ -31,37 +34,28 @@ namespace mid
...
-  void makeCluster(PreClusters::PreClusterBP& clBend, PreClusters::PreClusterBP& clBendNeigh, PreClusters::PreClusterNBP& clNonBend, const int& deIndex);
-
-  std::vector<Cluster2D> mClusters; ///< list of clusters
-  unsigned long int mNClusters = 0; ///< Number of clusters
+  bool loadPreClusters(gsl::span<const PreCluster>& preClusters);
+
+  bool makeClusters(PreClustersDE& pcs);
+  void makeCluster(const MpArea& areaBP, const MpArea& areaNBP, const int& icolumn, const int& deIndex);
+  void makeCluster(const PreClustersDE::BP& pcBP, const PreClustersDE::BP& pcBPNeigh, const PreClustersDE::NBP& pcNBP, const int& deIndex);
+  // void setPairedFlag(int icolumn, int ib, int value = 1);
```
Comment (aphecetche) [inline Detectors/MUON/MID/Clustering/include/MIDClustering/Clusterizer.h]: are those leftovers ?

File: Detectors/MUON/MID/Clustering/src/PreClusterHelper.cxx [hunk 2]
```diff
@@ -0,0 +1,66 @@
...
+  } else {
+    MpArea last = mMapping.stripByLocation(pc.lastStrip, pc.cathode, 0, pc.lastColumn, pc.deId, false);
+    lastX = last.getXmax();
+    lastY = (last.getYmax() > first.getYmax()) ? last.getYmax() : first.getYmax();
+    if (firstY > last.getYmin()) {
+      firstY = last.getYmin();
+    }
+  }
+  MpArea area(firstX, firstY, lastX, lastY);
+  return std::move(area);
```
Comment (aphecetche) [inline Detectors/MUON/MID/Clustering/src/PreClusterHelper.cxx]: why the explicit move here ?

File: Detectors/MUON/MID/Clustering/include/MIDClustering/Clusterizer.h (hunk 1 above)
Comment (dstocco) [reply Detectors/MUON/MID/Clustering/include/MIDClustering/Clusterizer.h]: Yep. Removed.

File: Detectors/MUON/MID/Clustering/src/PreClusterHelper.cxx (hunk 2 above)
Comment (dstocco) [reply Detectors/MUON/MID/Clustering/src/PreClusterHelper.cxx]: I thought that move was faster than copying the object.
However, MpArea is rather trivial and the compiler could optimise in a better way.
In the new amended version I'm returning the object instead.

**Pull request merged**